tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -355,54 +306,222 @@ def _process_resources(self, resources, event):
+
+            if container_states_tag_value:
+                for state in container_states_tag_value.split(';'):
+                    state_data = state.split(':')
+                    container_rid = state_data[0]
+                    container_throughput = state_data[1]
+
+                    container = next((c for c in resources if c['_rid'] == container_rid), None)
+
+                    if container:
+                        self._process_resource(container, account_client, container_throughput)
+            else:
+                self.log.warning('No tag {} on parent resource, {}.'.format(
+                    tag_name, parent_account))
+
+        except Exception as e:
+            log.warn(e)
+
+        return resources
+
+
+@CosmosDBCollection.action_registry.register('store-throughput-state')
+class CosmosDBStoreStateAction(AzureBaseAction):
+    """CosmosDB Store State Action
+
+    Stores the throughput of collections in a tag on the parent Cosmos DB account.
+    With accounts that have many collections, it's important to filter down which
+    collections to store since there is a tag length limit (approx 16 collections).
+
+    :example:
+
+    This policy stores the throughput of collections with throughput over 400 in
+    a tag called 'on-hour-state' on the parent Cosmos DB account.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: store-on-hours-state
+            resource: azure.cosmosdb-collection
+            filters:
+              - type: offer
+                key: content.offerThroughput
+                op: gt
+                value: 400
+            actions:
+              - type: store-throughput-state
+                state-tag: on-hour-state
+
+    """
+
+    schema = type_schema(
+        'store-throughput-state',
+        required=['state-tag'],
+        **{
+            'state-tag': {'type': 'string'}
+        }
+    )
+
+    TAG_VALUE_CHAR_LIMIT = 256
+
+    def _process_resources(self, resources, event):
+        OfferHelper.execute_in_parallel_grouped_by_account(
+            resources,
+            self.executor_factory,
+            self.manager.get_parent_manager(),
+            self._process_account_set,
+            self.log
+        )
+
+    def _process_account_set(self, resources, account_client):
+        account_tag_values = []
+        tag_name = self.data.get('state-tag')
+        cosmos_account = resources[0]['c7n:parent']
+
+        for resource in resources:
+            account_tag_values.append('{}:{}'.format(
+                resource['_rid'], resource['c7n:offer']['content']['offerThroughput']))
+
+        tag_value = ';'.join(account_tag_values)
+
+        if len(tag_value) > self.TAG_VALUE_CHAR_LIMIT:
+            raise('Can not add tag, {}, on parent resource, {}, '
+                  'because tag value exceeds allowed length.'
+                  'Add filters to reduce number of containers.'
+                  .format(tag_name, cosmos_account['name']))
```

Comment:
Thoughts on doing this check in the resource loop instead and just fitting in all you can, but also logging that collections were skipped?  

Just thinking that saving cost on `n` collections is better than `0`.   If you sorted `offerThroughput` and prioritized the largest ones it'd be even better.   

If they really have a lot of collections it may not be easy for them to fix this themselves, and they wouldn't have the ability to enforce the extremely useful sorting.

Reply:
So in my opinion they have the ability to filter by throughput so they can filter down to the collections with higher throughput. If they need to apply more filters they can filter by a naming convention, etc. 

It puts it in their power to decide which ones they want to impact rather than us deciding for them. 

I'm not sold this is the best way though. I'll think about it a little more.

Reply:
Yeah from the user perspective the people running the policies don't know much about collections though.  Like imagine being at Company A and having 900 subscriptions with 1500 CosmosDB accounts.  You just want to save money, tuning the policy (or making tons of policies) to work across your management group probably isn't possible. 

This is a net-new ask/enhancement though, I know its a bit of extra work, we don't have to decide or include it in this PR as I don't believe it is a breaking change to do this later after receiving customer feedback.